pymc_marketing/model_builder.py

```diff
@@ -291,22 +306,28 @@ def sample_model(self, **kwargs):
         --------
         >>> self.build_model()
         >>> idata = self.sample_model(draws=100, tune=10)
-        >>> assert isinstance(idata, xr.Dataset)
-        >>> assert "posterior" in idata
-        >>> assert "prior" in idata
-        >>> assert "observed_data" in idata
-        >>> assert "log_likelihood" in idata
         """
         if self.model is None:
             raise RuntimeError(
                 "The model hasn't been built yet, call .build_model() first or call .fit() instead."
             )
-
+        prior_predictive = kwargs.pop("prior_predictive", False)
+        posterior_predictive = kwargs.pop("posterior_predictive", False)
         with self.model:
             sampler_args = {**self.sampler_config, **kwargs}
             idata = pm.sample(**sampler_args)
-            idata.extend(pm.sample_prior_predictive())
-            idata.extend(pm.sample_posterior_predictive(idata))
+            if prior_predictive:
+                if prior_kwargs is not None:
```

Comment:
Nitpick: I would create an empty dict, so the same call to `pm.sample_prior_predictive` can be done without repeating code